src/structures/CoffeeQueue.ts

```diff
@@ -61,9 +61,9 @@ export class CoffeeQueue extends Array<CoffeeTrack | UnresolvedTrack> {
   }
 
   /** Progress to next song */
-  public progress(): CoffeeTrack | UnresolvedTrack | undefined {
+  public progress(): CoffeeTrack<T> | UnresolvedTrack<T> | undefined {
     this.previous = this.current
-    this.current = this.shift() as CoffeeTrack
+    this.current = this.shift()
```

Comment:
Btw I removed this cast because my IDE was not complaining after doing so, Idk if it does for you